### PR TITLE
Adjust minor SAST warnings

### LINKED ISSUE #3419: Minor SAST warnings
Hello! I have run https://svace.pages.ispras.ru/svace-website/en/ against neo-go codebase and here are some of the issues found. I think most of them are OK (I ignored issues such as having empty set of oracles in the oracle post-persist)

After discussion of @roman-khimov, I decided to create it here, both seems minor.

1. Here prevHeader could be nil in the else branch.
```
Message: After having been compared to a nil value at blockchain.go:2984, pointer 'prevHeader' is dereferenced at blockchain.go:2987.
```
https://github.com/nspcc-dev/neo-go/blob/a13dc59bd914aec008c842e147c9fe71f9988504/pkg/core/blockchain.go#L2984

2. Here an attacker could use NeoFS protocol scheme to cause panic in production node without neofs support
```
Variable len(t67.MainCfg.NeoFS.Nodes), whose possible value set allows a zero value at request.go:162 by calling function 'builtin.len', is used as a denominator at request.go:162.
```
https://github.com/nspcc-dev/neo-go/blob/a13dc59bd914aec008c842e147c9fe71f9988504/pkg/services/oracle/request.go#L163

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Close #3419

### CODE REVIEW

File: pkg/core/blockchain.go [hunk 1]
```diff
@@ -3018,8 +3018,12 @@ func (bc *Blockchain) verifyTxWitnesses(t *transaction.Transaction, block *block
 // verifyHeaderWitnesses is a block-specific implementation of VerifyWitnesses logic.
 func (bc *Blockchain) verifyHeaderWitnesses(currHeader, prevHeader *block.Header) error {
 	var hash util.Uint160
-	if prevHeader == nil && currHeader.PrevHash.Equals(util.Uint256{}) {
-		hash = currHeader.Script.ScriptHash()
+	if prevHeader == nil {
```
Comment (roman-khimov) [inline pkg/core/blockchain.go:3021]: It's never `nil`.

File: pkg/core/blockchain.go [hunk 2]
```diff
@@ -3018,7 +3018,7 @@ func (bc *Blockchain) verifyTxWitnesses(t *transaction.Transaction, block *block
 // verifyHeaderWitnesses is a block-specific implementation of VerifyWitnesses logic.
 func (bc *Blockchain) verifyHeaderWitnesses(currHeader, prevHeader *block.Header) error {
 	var hash util.Uint160
-	if prevHeader == nil && currHeader.PrevHash.Equals(util.Uint256{}) {
+	if currHeader.PrevHash.Equals(util.Uint256{}) {
 		hash = currHeader.Script.ScriptHash()
```
Comment (AnnaShaleva) [inline pkg/core/blockchain.go:3022]: It seems like a dead code to me. @roman-khimov, do we really need this path? Also, verifying against the block's own Script's hash makes almost no sense.

File: pkg/services/oracle/request.go [hunk 3]
```diff
@@ -157,6 +157,11 @@ func (o *Oracle) processRequest(priv *keys.PrivateKey, req request) error {
 				resp.Code = transaction.Error
 			}
 		case neofs.URIScheme:
+			if len(o.MainCfg.NeoFS.Nodes) == 0 {
+				o.Log.Warn("No NeoFS nodes configured", zap.String("url", req.Req.URL))
```
Comment (AnnaShaleva) [inline pkg/services/oracle/request.go:161]: Log should start with a lowercase letter.

File: pkg/core/blockchain.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/core/blockchain.go:3022]: Genesis block never faces this code anyway.

Comment (roman-khimov) [reply pkg/core/blockchain.go:3022]: This case can be valid only for genesis. Genesis block never runs this code. Throw it away at last, please.

File: pkg/core/blockchain.go [hunk 4]
```diff
@@ -3015,18 +3016,6 @@ func (bc *Blockchain) verifyTxWitnesses(t *transaction.Transaction, block *block
...
-	var hash util.Uint160
-	if prevHeader == nil && currHeader.PrevHash.Equals(util.Uint256{}) {
-		hash = currHeader.Script.ScriptHash()
-	} else {
-		hash = prevHeader.NextConsensus
-	}
-	_, err := bc.VerifyWitness(hash, currHeader, &currHeader.Script, HeaderVerificationGasLimit)
-	return err
-}
-
```
Comment (AnnaShaleva) [inline pkg/core/blockchain.go:3029]: Let's keep this method, just remove the useless `if`.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**